File: CodingSample/DotNet/Fundamentals/Inheritance/InterfaceTypeTest/BankLib/CurrentAccount.cs
namespace Banking;

//a class defined with 'sealed' keyword cannot be extended
sealed class CurrentAccount : Account
{
    public override void Withdraw(decimal amount)
    {
        Balance -= amount;
    }

    public override void Deposit(decimal amount)
    {
        if(Balance < 0)
            amount *= 0.9m;
        Balance += amount;
    }
}

File: CodingSample/Java/Core/Inheritance/InterfaceTest/bank/banking/CurrentAccount.java
package banking;

//a class declared with 'final' keyword cannot be extended
final class CurrentAccount extends Account {
    
    public void deposit(double amount) {
        if(balance < 0)
            amount *= 0.9;
        balance += amount;
    }

    public void withdraw(double amount) throws InsufficientFundsException {
        balance -= amount;
    }
}
